extension/js/common/composer.ts

```diff
@@ -1149,58 +1163,43 @@ export class Composer {
-    const result = await PgpMsg.decrypt({ kisWithPp: await Store.keysGetAllWithPassphrases(this.urlParams.acctEmail), encryptedData: Buf.fromUtfStr(armoredMsg) });
-    if (result.success) {
-      if (!Mime.resemblesMsg(result.content)) {
-        this.appendForwardedMsg(result.content);
-      } else {
-        const mimeDecoded = await Mime.decode(result.content);
-        if (typeof mimeDecoded.text !== 'undefined') {
-          this.appendForwardedMsg(Buf.fromUtfStr(mimeDecoded.text));
-        } else if (typeof mimeDecoded.html !== 'undefined') {
-          this.appendForwardedMsg(Buf.fromUtfStr(Xss.htmlSanitizeAndStripAllTags(mimeDecoded.html!, '\n')));
-        } else {
-          this.appendForwardedMsg(result.content);
-        }
-      }
-    } else {
-      Xss.sanitizeAppend(this.S.cached('input_text'), `<br/>\n<br/>\n<br/>\n${armoredMsg.replace(/\n/g, '<br/>\n')}`);
-    }
+  private generateHTMLRepliedPart = (text: string, date: Date, from: string) => {
+    const header = `On ${this.formatDate(date)} ${from} wrote:<br>`;
+    return header + this.quoteText(Xss.escape(text));
   }
 
-  private renderReplyMsgComposeTable = async (method: 'forward' | 'reply' = 'reply') => {
+  private renderReplyMsgComposeTable = async (method: 'forward' | 'reply' = 'reply'): Promise<void> => {
     this.S.cached('prompt').css({ display: 'none' });
     this.S.cached('input_to').val(this.urlParams.to.join(',') + (this.urlParams.to.length ? ',' : '')); // the comma causes the last email to be get evaluated
     await this.renderComposeTable();
     if (this.canReadEmails) {
-      const determined = await this.app.emailProviderDetermineReplyMsgHeaderVariables();
-      if (determined && determined.lastMsgId && determined.headers) {
-        this.additionalMsgHeaders['In-Reply-To'] = determined.headers['In-Reply-To'];
-        this.additionalMsgHeaders.References = determined.headers.References;
+      if (this.messageToReplyOrForward && this.messageToReplyOrForward.text) {
         if (method === 'forward') {
+          this.additionalMsgHeaders['In-Reply-To'] = this.messageToReplyOrForward.headers.inReplyToHeaders;
+          this.additionalMsgHeaders.References = this.messageToReplyOrForward.headers.references;
           this.urlParams.subject = 'Fwd: ' + this.urlParams.subject;
-          await this.retrieveDecryptAddForwardedMsg(determined.lastMsgId);
+          await this.appendForwardedMsg(this.messageToReplyOrForward.text);
+        } else {
+          if (!this.messageToReplyOrForward.headers.from || !this.messageToReplyOrForward.headers.date) {
+            return;
+          }
+          const sentDate = new Date(String(this.messageToReplyOrForward.headers.date));
+          this.msgExpandingHTMLPart = '<br><br>' + this.generateHTMLRepliedPart(this.messageToReplyOrForward.text, sentDate, this.messageToReplyOrForward.headers.from);
+          if (!this.urlParams.draftId) {
+            this.showExpandButton(this.msgExpandingHTMLPart);
```

Comment:
No need to resize compose box here?
(I'm not sure)

Reply:
resizeComposeBox is below, after `else` statement

Reply:
Right, but that one will only run if you enter the `else` branch. How about in this `if` branch evaluates true?